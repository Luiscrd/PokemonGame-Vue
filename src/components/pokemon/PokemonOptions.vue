<template>
    <div class="options-container mt-5 mb-4">
        <ul>
            <button
            v-for="pokemon in pokemons"
            :key="pokemon.id"
            type="button"
            class="btn btn-primary"
            @click="$emit('selection', pokemon.id)"
            >{{ pokemon.name }}</button>
        </ul>
    </div>
</template>
  
<script>
  export default {
    props: {

        pokemons: {
            type: Array,
            required: true
        }

    }
  }
</script>
  
<style scoped>
ul {
    list-style-type: none;
    margin-left: 0;
    padding-left: 0;
}
li {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgba(0, 0, 0, 0.2);
    cursor: pointer;
    margin-bottom: 10px;
    width: 250px;
}

li:hover {
    background-color: rgba(0, 0, 0, 0.05);
}

.btn:hover {
    color: white;
    background-color: #13191c;
    border-color: #13191c;
}
.btn-primary {
    --bs-btn-color: #fff;
    --bs-btn-bg: #263238;
    --bs-btn-border-color: #263238;
    --bs-btn-hover-color: #fff;
    --bs-btn-hover-bg: #263238;
    --bs-btn-hover-border-color: #263238;
    --bs-btn-focus-shadow-rgb: 49,132,253;
    --bs-btn-active-color: #fff;
    --bs-btn-active-bg: #263238;
    --bs-btn-active-border-color: #263238;
    --bs-btn-active-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
    --bs-btn-disabled-color: #fff;
    --bs-btn-disabled-bg: #263238;
    --bs-btn-disabled-border-color: #263238;
    width: 100%;
    margin-bottom: 10px;
}


.options-container {
    display: flex;
    justify-content: center;
}
  </style>