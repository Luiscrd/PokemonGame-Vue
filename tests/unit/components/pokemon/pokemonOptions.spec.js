import { shallowMount } from '@vue/test-utils'
import PokemonOptions from '@/components/pokemon/PokemonOptions.vue'
import { mockPokemons } from '../../mocks/pokemons.mock'

describe('PokemonOptions Component', () => {

    let wrapper;

    beforeEach(() => {

        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemons: mockPokemons
            }
        })

    });

    test('Debe de hacer mach con el snapshot', () => {

        console.log(wrapper.html());

        expect(wrapper.html()).toMatchSnapshot()

    })

    test('Debe de mostrar las 4 opciones correctamente', () => {

        const btns = wrapper.findAll('button')

        expect( btns.length).toBe(4)

        const [btn1, btn2, btn3, btn4] = btns

        expect(btn1.text()).toBe('bulbasaur')

        expect(btn2.text()).toBe('ivysaur')

        expect(btn3.text()).toBe('venusaur')

        expect(btn4.text()).toBe('charmander')

    })

    test('Debe emitir "selection" con sus respectivos parametros al hacer click', () => {

        const [btn1, btn2, btn3, btn4] = wrapper.findAll('button')

        btn1.trigger('click')

        btn2.trigger('click')

        btn3.trigger('click')

        btn4.trigger('click')

        expect( wrapper.emitted('selection').length ).toBe(4)

        expect( wrapper.emitted('selection')[0] ).toEqual([1])

        expect( wrapper.emitted('selection')[1] ).toEqual([2])

        expect( wrapper.emitted('selection')[2] ).toEqual([3])
        
        expect( wrapper.emitted('selection')[3] ).toEqual([4])

    })

})