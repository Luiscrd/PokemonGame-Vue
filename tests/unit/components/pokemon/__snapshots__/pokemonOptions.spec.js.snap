// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component Debe de hacer mach con el snapshot 1`] = `
<div class="options-container mt-5 mb-4">
  <ul>
    <li><button type="button" class="btn btn-primary">bulbasaur</button></li>
    <li><button type="button" class="btn btn-primary">ivysaur</button></li>
    <li><button type="button" class="btn btn-primary">venusaur</button></li>
    <li><button type="button" class="btn btn-primary">charmander</button></li>
  </ul>
</div>
`;
